.gitignore

```diff
@@ -200,3 +200,7 @@ launchSettings.json
 
 .idea
 .ionide
+
+localize
+MicroBuild
+*.out
```

Comment:
Regarding `localize`, is there a reason we can't put/keep this under `artifacats/` which is already gitignored? With this change, if anyone runs the tool to create the files locally, but then checks out an old release branch, all those ignored files are no longer going to be ignored on the release branches and will show as changes.

I don't see anything in the PR to hint why files are being created under a `MicroBuild` directory (or why a `MicroBuild` file is being created).

Similarly, I don't know what's creating `*.out` files, or why it's doing it outside the `artifacts/`, or `obj/` folders. I can't think of any reason we'd want to commit a `.out` file into our repo, but `*.out` is not a well known temporary file extension like binlog, suo.

Reply:
> Regarding localize, is there a reason we can't put/keep this under artifacats/ which is already gitignored? With this change, if anyone runs the tool to create the files locally, but then checks out an old release branch, all those ignored files are no longer going to be ignored on the release branches and will show as changes.

That is to avoid checking in all NuGet.Build.Localization into `.\localize` repo. Sorry for the name clashing. I can change it for something more clear if you want.

`.\artifacts\localize` is a path that MicroBuild Localization plug-in uses to store results.

> I don't see anything in the PR to hint why files are being created under a MicroBuild directory (or why a MicroBuild file is being created).

That's the path the CI choses to install [MicroBuild Localization plug-in](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4667107&view=logs&j=2ea8c801-c4c3-578e-0ff0-bcf4c12d9404&t=a09330d5-95f5-5056-3c03-30e818af5d97&l=9). I added that `.gitignore` to avoid checking in my own MicroBuild local setup.

> Similarly, I don't know what's creating *.out files, or why it's doing it outside the artifacts/, or obj/ folders. I can't think of any reason we'd want to commit a .out file into our repo, but *.out is not a well known temporary file extension like binlog, suo.

I personally use `*.out` to store command output into a file:

```
.\build.ps1 *> build.out
```